src/glfw/window.cc

```diff
@@ -0,0 +1,239 @@
+#include <functional>
+#include <iostream>
+#include <iterator>
+#include <memory>
+#include <string>
+
+#include <GLFW/glfw3.h>
+#include <spdlog/spdlog.h>
+
+#include "window/error.h"
+#include "window/window.h"
+
+using namespace wnd;
+
+namespace
+{
+
+void
+setWindowHints()
+{
+    glfwDefaultWindowHints();
+
+    // clang-format off
+    glfwWindowHint( GLFW_RESIZABLE,               GLFW_TRUE );
+    glfwWindowHint( GLFW_VISIBLE,                 GLFW_TRUE );
+    glfwWindowHint( GLFW_FOCUSED,                 GLFW_TRUE );
+    glfwWindowHint( GLFW_FLOATING,                GLFW_FALSE );
+    glfwWindowHint( GLFW_TRANSPARENT_FRAMEBUFFER, GLFW_FALSE );
+    glfwWindowHint( GLFW_FOCUS_ON_SHOW,           GLFW_TRUE );
+    glfwWindowHint( GLFW_DOUBLEBUFFER,            GLFW_TRUE );
+    glfwWindowHint( GLFW_REFRESH_RATE,            GLFW_DONT_CARE );
+    glfwWindowHint( GLFW_CLIENT_API,              GLFW_NO_API );
+    // clang-format on
+} // setWindowHints
+
+std::string
+getHandleAddressString( const Window* window )
+{
+    return std::to_string( std::bit_cast<uintptr_t>( window ) );
```

Comment:
Here you use ```std::bit_cast``` but the required header <bit> is not included.
https://en.cppreference.com/w/cpp/numeric/bit_cast